did-tezos/src/lib.rs

```diff
@@ -130,16 +201,162 @@ impl DIDMethod for DIDTz {
+        Ok(explorer::execute_service_view(did, &did_manager, network).await?)
+    }
+
+    async fn tier3_updates(
+        &self,
+        prefix: &str,
+        doc: &mut Document,
+        updates: Updates,
+        public_key: Option<String>,
+    ) -> Result<()> {
+        match updates {
+            Updates::SignedIetfJsonPatch(patches) => {
+                for jws in patches {
+                    let mut doc_json = serde_json::to_value(&mut *doc)?;
+                    let curve = prefix_to_curve_type(prefix)
+                        .ok_or(anyhow!("Unsupported curve."))?
+                        .0
+                        .to_string();
+                    let jwk = match prefix {
+                        "tz1" => {
+                            let pk = match public_key {
+                                Some(ref p) => {
+                                    PublicKey::from_base58check(&p.clone()).unwrap().as_ref()[..]
+                                        .to_vec()
+                                }
+                                None => return Err(anyhow!("Need public key for signed patches")),
+                            };
+                            JWK {
+                                params: Params::OKP(OctetParams {
+                                    curve,
+                                    public_key: Base64urlUInt(pk),
+                                    private_key: None,
+                                }),
+                                public_key_use: None,
+                                key_operations: None,
+                                algorithm: None,
+                                key_id: None,
+                                x509_url: None,
+                                x509_thumbprint_sha1: None,
+                                x509_certificate_chain: None,
+                                x509_thumbprint_sha256: None,
+                            }
+                        }
+                        "tz2" => {
+                            // TODO use tezedge_client when it handles tz2
+                            let pk = match public_key {
+                                Some(ref p) => p.from_base58check().unwrap()[4..].to_vec(),
+                                None => return Err(anyhow!("Need public key for signed patches")),
+                            };
+                            JWK {
+                                // TODO I don't think the coordinates mean anything in compressed
+                                // form
+                                params: Params::EC(ECParams {
+                                    curve: Some("secp256k1".to_string()),
+                                    x_coordinate: Some(Base64urlUInt(pk[0..17].to_vec())),
+                                    y_coordinate: Some(Base64urlUInt(pk[17..33].to_vec())),
+                                    ecc_private_key: None,
```

Comment:
The compressed form is the `x` coordinate and a sign bit. The `y` coordinate can be reconstructed from those. I think the x coordinate here would be `pk[1..33]` (after checking `pk[0]`) and `y` left empty, rather than splitting them at byte `17`. But RFC 8812 says [compressed encoding must be used for EC JWK](https://tools.ietf.org/html/rfc8812#section-3.1), so I have been assuming that the JWK should always include both `x` and `y` coordinate for `secp256k1`.

In `did-key` we construct a JWK (in uncompressed form) given the compressed serialization, in function [`secp256k1_parse`](https://github.com/spruceid/ssi/blob/main/did-key/src/lib.rs#L28-L51). That could maybe be moved into `src/jwk.rs` so it could be reused here. It uses [`secp256k1::PublicKey::parse_slice`](https://docs.rs/libsecp256k1/0.3.5/secp256k1/struct.PublicKey.html#method.parse_slice) and then [`secp256k1::PublicKey::serialize`](https://docs.rs/libsecp256k1/0.3.5/secp256k1/struct.PublicKey.html#method.serialize) via [`ECParams::try_from<&secp256k1::PublicKey>`](https://github.com/spruceid/ssi/blob/main/src/jwk.rs#L617-L632).

Reply:
Thank you very much, fixed in 3f8e46f